core/arch/arm/mm/core_mmu_v7.c

```diff
@@ -671,8 +671,10 @@ static void map_memarea(struct tee_mmap_region *mm, uint32_t *ttb)
 	 * TODO: support mapping devices at a virtual address which isn't
 	 * the same as the physical address.
 	 */
-	if (mm->va < (NUM_UL1_ENTRIES * SECTION_SIZE))
-		panic("va conflicts with user ta address");
+	if (ttb == (void *)core_mmu_get_main_ttb_va()) {
```

Comment:
When suspended, DRAM is in self refresh state and can not be accessible. So need to build a mmu table in IRAM and mapping devices that needed when suspending and resuming. The mapping is based on 1M side, not based on page, because IRAM is small and page mapping is not allowed. Now when I use map_memarea, I pass 1M aligned address to it and it can map it based on section mapping.